### PR TITLE
DOCS-11425-aggregators-studio-configuration-kt

### PR DESCRIPTION
**Pull request by kevintroller**
New Studio configuration page for Aggregators Module

- Moved Configure Object Store section from the User Guide to the Studio page.
- Reworded
- Added Studio steps and screenshots

### CODE REVIEW

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+. <<create-mule-project,Create a Mule project>>.
+. <<add-connector-to-project,Add the module to your Mule project>>.
+. <<configure-source,Configure a source for the module's flow>>.
+. <<add-connector-operation,Add an Aggregator Scope to the flow>>.
+. <<configure-os-aggregator,Configure an Object Store for an Aggregator>>.
+
+When you run the connector, you can view the app log to check for problems in real time, as described in <<view-app-log,View the App Log>>.
+
+If you are new to configuring connectors in Studio, see xref:connectors::introduction/intro-config-use-studio.adoc[Use Studio to Configure a Connector]. If, after reading this topic, you need additional information about the connector fields, see the <link to the connector Reference Guide [<connector name> Reference].
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]: links TBD?

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+A source initiates a flow when a specified condition is met.
+You can configure one of these sources to use with Aggregators Module:
+
+* *HTTP > Listener*
++
+Initiates a flow each time it receives a request on the configured host and port.
+* *Aggregator listener*
++
+Initiates a sub-flow by listening to elements triggered by a referenced aggregator scope. +
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
Initiates a subflow by listening to elements triggered by a referenced aggregator scope. +
```

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+After the aggregator scope referenced by the listener completes an aggregation, the *Aggregator listener* triggers the sub-flow with a list of all the aggregated elements.
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
After the aggregator scope referenced by the listener completes an aggregation, *Aggregator listener* triggers the subflow with a list of all the aggregated elements.
```
Are you sure about using bold here? We're essentially saying that the label triggers the subflow....

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,148 @@
+= Use Studio to Configure Aggregators Module 1.0 - Mule 4
+
+Anypoint Studio (Studio) editors help you design and update your Mule applications, properties, and configuration files.
+
+To add and configure a module in Studio:
+
+. <<create-mule-project,Create a Mule project>>.
+. <<add-connector-to-project,Add the module to your Mule project>>.
+. <<configure-source,Configure a source for the module's flow>>.
+. <<add-connector-operation,Add an Aggregator Scope to the flow>>.
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
. <<add-connector-operation,Add an aggregator scope to the flow>>.
```

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+. <<configure-os-aggregator,Configure an Object Store for an Aggregator>>.
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
. <<configure-os-aggregator,Configure an object store for an aggregator>>.
```

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+Adding a module to a Mule project in Studio does not make that module available to other projects in your Studio workspace.
+
+
+[[configure-source]]
+== Configure a Source
+
+A source initiates a flow when a specified condition is met.
+You can configure one of these sources to use with Aggregators Module:
+
+* *HTTP > Listener*
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc:46]: The name includes >?

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+Initiates a sub-flow by listening to elements triggered by a referenced aggregator scope. +
+After the aggregator scope referenced by the listener completes an aggregation, the *Aggregator listener* triggers the sub-flow with a list of all the aggregated elements.
+
+For example, to configure an Aggregator listener, follow these steps:
+
+. In the *Mule Palette* view, select *Aggregators > Aggregator listener*.
+. Drag *Aggregator listener* to the Studio canvas.
+. On the *Aggregator listener* configuration screen, optionally change the value of the *Display Name* field.
+. Set *Aggregator name* to the name of the aggregator scope to listen to. +
+The listener executes after the aggregator scope releases its elements.
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
. Set *Aggregator name* to the name of the aggregator scope to listen to. +
+
The listener executes after the aggregator scope releases its elements.
```

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+. Select *Include timed out groups* to indicate whether the listener should be triggered when a group is released due to a timeout. +
+This configuration option also works for non-group aggregators, such as the *Time based aggregator* and *Size based aggregator* scopes.
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
. Select *Include timed out groups* to indicate whether the listener should be triggered when a group is released due to a timeout. +
+
This configuration option also works for nongroup aggregators, such as the *Time based aggregator* and *Size based aggregator* scopes.
```

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+To add a scope for Aggregators Module, follow these steps:
+
+. In the *Mule Palette* view, select *Aggregators* and then select the desired scope:
++
+* *Group based aggregator* +
+Enables you to aggregate elements into groups by group ID.
+* *Size based aggregator* +
+Enables you to aggregate elements until a predefined size number of elements completes the aggregation.
+* *Time based aggregator* +
+Enables you to aggregate elements within a predefined time limit.
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
* *Group based aggregator*: Enables you to aggregate elements into groups by group ID.
* *Size based aggregator*: Enables you to aggregate elements until a predefined number of elements completes the aggregation.
* *Time based aggregator*: Enables you to aggregate elements within a predefined time limit.
```
OR
```suggestion
* *Group based aggregator* +
+
Enables you to aggregate elements into groups by group ID.
* *Size based aggregator* +
+
Enables you to aggregate elements until a predefined size number of elements completes the aggregation.
* *Time based aggregator* +
+
Enables you to aggregate elements within a predefined time limit.
```

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
++
+. Drag the scope onto the Studio canvas to the right of the source.
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
. Drag the scope onto the Studio canvas to the right of the source:
```

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+----
+<aggregators:group-based-aggregator name="globalOSAggregator"
+                                   maxSize="10"
+                                   objectStore="aGlobalObjectStore">
+
+----
+
+=== Configure a Private Object Store
+
+To configure a private object store in Studio, you need to manually update the XML code:
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc]:
```suggestion
To configure a private object store in Studio, manually update the XML code:
```

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+    <aggregators:object-store>
+        <os:private-object-store alias="privateObjectStore" persistent="false"/>
+    </aggregators:object-store>
+</aggregators:group-based-aggregator>
+----
+
+[[view-app-log]]
+== View the App Log
+
+To check for problems, you can view the app log as follows:
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc:141]: Although checking for problems is what they are doing, I wonder if we should avoid implying that problems will undoubtedly occur or are likely?
How about...:
```suggestion
To monitor app performance, view the app log as follows:
```

Comment (kbostrom) [reply aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc:141]: If this ^ is part of the template, let's discuss at a team meeting.

File: aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+
+* If you’re running the app from Anypoint Platform, the app log output is visible in the Anypoint Studio console window.
+* If you’re running the app using Mule from the command line, the app log output is visible in your OS console.
```
Comment (kbostrom) [inline aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc:144]: It also occurs to me that we can make these less passive:
```suggestion
* If you’re running the app from Anypoint Platform, view app log output in the Anypoint Studio console window.
* If you’re running the app using Mule from the command line, view app log output in your OS console.
```

Comment (kbostrom) [reply aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc:144]: Ditto re. template ^

Comment (kbostrom) [reply aggregators/1.0/modules/ROOT/pages/aggregators-studio-configuration.adoc:144]: The previous rewordings then are parallel to the following sentence: you can also view

**Review by kbostrom — DISMISSED** [review]
CE, CU

**Review by giss-ignacio — APPROVED** [review]

**Review by dhirsch1 — APPROVED** [review]

**Pull request merged**